supervisor/api/backups.py

```diff
@@ -503,6 +504,14 @@ async def upload(self, request: web.Request):
             if location and location != LOCATION_CLOUD_BACKUP:
                 tmp_path = location.local_where
 
+        filename: str | None = None
+        if ATTR_FILENAME in request.query:
+            filename = request.query.get(ATTR_FILENAME)
+            try:
+                vol.Match(RE_BACKUP_FILENAME)(filename)
```

Comment:
Now that we have a couple places we use query string for input we probably should modify `api_validate` to support passing a schema to use against the query string. Refactoring effort for later